### PR TITLE
Add missing test for send letter job

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Test
+    public void should_NOT_process_message_if_ftp_is_not_available() {
+        given(availabilityChecker.isFtpAvailable(any())).willReturn(false);
+        SendLetterJob job = new SendLetterJob(messageProcessor, availabilityChecker);
+
+        job.run();
+
+        verify(messageProcessor, never()).process();
+    }
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java:29]: could you please add `verifyNoMoreInteractions`?

File: src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Test
+    public void should_process_message_if_ftp_is_available() {
+        given(availabilityChecker.isFtpAvailable(any())).willReturn(true);
+        SendLetterJob job = new SendLetterJob(messageProcessor, availabilityChecker);
+
+        job.run();
+
+        verify(messageProcessor, times(1)).process();
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java:38]: could you please add verifyNoMoreInteractions?

File: src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SendLetterJobTest {
+
+    @Mock private MessageProcessor messageProcessor;
+    @Mock private FtpAvailabilityChecker availabilityChecker;
+
+    @Test
+    public void should_NOT_process_message_if_ftp_is_not_available() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java]: NOT can be in lower case :)

Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java]: it's intentionally uppercase :)

File: src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/slc/services/SendLetterJobTest.java:29]: I'd rather not.
https://static.javadoc.io/org.mockito/mockito-core/2.13.0/org/mockito/Mockito.html#finding_redundant_invocations

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**